qt/scientific_interfaces/Muon/IALCDataLoadingPresenterSubscriber.h

```diff
@@ -0,0 +1,27 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2024 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#pragma once
+
+#include "DllConfig.h"
+
+namespace MantidQt {
+namespace CustomInterfaces {
+
+/** IALCDataLoadingPresenterSubscriber : Subscriber's interface for getting updates from
+ * ALCDataLoadingPresenter
+ */
+class MANTIDQT_MUONINTERFACE_DLL IALCDataLoadingPresenterSubscriber {
```

Comment:
According to the pattern these have been implemented(in other similar places as well), something like `IALCDataLoadingPresenterCallback` could have been a better name for this class - because it has always been a single subscriber working behind the scenes. But I think now its too late for that renaming.

Reply:
Yes I agree, this might be a future maintenance task for the muon area

Reply:
Yeah since functionally this is absolutely fine, lets mark it as a maintenance task if you are happy. Thanks!